Add a new Chobby skin + icon refresh

completesomething opened the pull request:
This PR reskins Chobby to be a bit more modern, using the Evolved skin as a base.

For now, every skin-related change is currently stored within the Evolved skin to easily rebase and check what exactly has been changed. When the PR is ready, these changes will moved to a new skin.

 **DO NOT MERGE UNTIL THEY'RE MOVED TO A NEW SKIN FOLDER** 

Feel free to nitpick or make suggestions.

Preview:

<img width="1920" height="1080" alt="439466588-002c1df4-c162-4313-b409-8f09b6e78b18" src="https://github.com/user-attachments/assets/4a7b4804-e7f4-4c73-9d39-24237e8d6c33" />

See https://github.com/ZeroK-RTS/Chobby/assets/76411090/c561c699-4f7f-4faf-9a22-3307ae4dffe6 for old preview

completesomething committed:
Colorize battle buttons based on their states

completesomething commented:
Elements of this new skin could possibly be reused for a future in-game UI reskin.

There was originally a plan to have a drop shadow for the main two large panels, but this plan was dropped for consistency and somewhat technical reasons, as the floating buttons on the left and top-right would need drop shadows as well. A drop shadow shader, if possible, would be the cleanest solution.

GoogleFrog commented:
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/0cae0a93-f9db-418d-9544-f5f14fd11126)
Looks good. Just on this screenshot.
- Should the battle list have the same border as the windows on the right?
- The border colour seems a bit bright to me, at least in the lower right regions where it is contrasting with the dark part of the gradient. A gradient on the border doesn't seem good or viable. Maybe the gradient is too strong or the borders could be tweaked less bright,
- Is the host game and discord button the same colour? The bottom buttons seems a bit too accentuated?
- The lobby news headings have a darker background, which seems bad.
- Filter out and password appear to have misaligned.
- The bottom right of the battle list window gets quite dark.

On sources.
- The license has to be pretty permissive because this sort of stuff needs ending in the future.
- Sources etc need to be committed because otherwise the lobby art becomes write-only.

completesomething commented:
>     * Should the battle list have the same border as the windows on the right?

If you're referring to the 1-pixel border, I'll investigate.

>     * The border colour seems a bit bright to me, at least in the lower right regions where it is contrasting with the dark part of the gradient. A gradient on the border doesn't seem good or viable. Maybe the gradient is too strong or the borders could be tweaked less bright,

Sure. I've tweaked the main panel gradient to be very subtle now:

![preview2](https://github.com/ZeroK-RTS/Chobby/assets/76411090/5a4d4dda-0b54-4e67-a20d-db9da53a59ed)

Currently uncommitted.

>     * Is the host game and discord button the same colour? The bottom buttons seems a bit too accentuated?

They're the same color/class. What you're observing might be due to the bright color of the Discord button contrasting with the darker gradient of the background.

>     * The lobby news headings have a darker background, which seems bad.

One of those loose ends. I'll look into it.

>     * Filter out and password appear to have misaligned.

This is due to the font shadow for some reason. I'll probably remove the font shadows for that kind of text.

>     * The bottom right of the battle list window gets quite dark.

Check above screenshot.

>     * The license has to be pretty permissive because this sort of stuff needs ending in the future.

Sure.

>     * Sources etc need to be committed because otherwise the lobby art becomes write-only.

For icons and the main panels, sure. For the buttons, some source files will be missing because I didn't bother saving much sources during the experimentation phase.

GoogleFrog commented:
Is this supposed to be WIP still?

GoogleFrog commented:
To clarify, I don't tend to look at or review WIPs in much detail. When it is done enough to not be "WIP" I poke through it more thoroughly and probably clone it locally.

completesomething commented:
> Is this supposed to be WIP still?

There are still things I want to fix, yes.

completesomething commented:
The icon refresh part of work is basically finished within the scope I was planning, but I haven't licensed them yet.

completesomething commented:
This PR is ready for review now.

For reference, current Evolved:

![currentEvolved](https://github.com/ZeroK-RTS/Chobby/assets/76411090/21b403bf-b421-4734-90e3-07720079f378)

My revised Evolved:

![myEvolved](https://github.com/ZeroK-RTS/Chobby/assets/76411090/eb04a0fa-0a93-469e-b488-28505cffab7d)

Do not merge until these are changes are moved to a new skin folder.

> Should the battle list have the same border as the windows on the right?

I think it's fine. The battle buttons and their layout serve a different purpose, and the welcome screen looks bare without the borders.

> The lobby news headings have a darker background, which seems bad.

I think this is fine. They help indicate that they are link buttons, and they also serve as faux separators, so the headings would look bare without it.

GoogleFrog commented:
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/a703083f-d571-457c-8fc4-522335694edf)
Chat tabs in single panel mode are missing background.

GoogleFrog commented:
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/5cf4a61a-5ffc-4b53-9a99-e474ba195fa4)
Chat text box colour isn't the same as the boxes that hold player names and chat. Battle room and chat channels.
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/89186e2e-3d75-425b-b87d-abbc2d51dc2e)
Idk if this is inherently bad, but seems like a weird/accidental choice at least.

GoogleFrog commented:
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/b6a3ccdc-d030-4ea0-a32d-cb679dbd962c)
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/66335a37-251e-447c-a4af-9e45e24b9efe)
Do clickable things always need to be dark like this? I can see the argument. But it makes the surrounding grey look a bit off/dirty.

GoogleFrog commented:
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/1cab8e8a-eadc-44af-919c-943993c6002b)
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/00415075-bb91-450c-8a3c-175ff54efff2)
The grey in general seems to make things a bit messier/harder to read. It has lower contrast.

GoogleFrog commented:
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/57ed3621-c090-4537-a4d6-5748a9d8c9bd)
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/a1b39452-5013-4dc9-9b2a-d9f081d35839)
How thoroughly was the increased width for these buttons tested? I could see the testing being a pain.

![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/ee28df6c-0fbe-4222-998b-baadb773cd58)
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/ac2b6909-f22a-4feb-9053-31dc9a5e819b)
I see the change was a bit more than just a scale. What was it and why?

GoogleFrog commented:
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/a63be814-ae00-41dd-aa53-94471dd9bfe9)
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/478686a4-5a38-4e47-9187-78e18bbb8bc4)
The top battle status is too blue when the battle is selected. Actually this feedback probably extends to the main menu buttons as well. I am not sure about having a blue box sit on the edge of the screen all the time. Seems a bit noisy.

GoogleFrog commented:
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/b85105f3-3165-4db8-91fe-96ee644110ea)
The tickboxes seem ambiguous. Looking at it twice this feedback surprises me. But when I first looked at this page i just got a feeling of not knowing whether the box was selected or not.

GoogleFrog commented:
> I see the change was a bit more than just a scale. What was it and why?
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/32382ff6-777b-458f-a0ae-8b06a29d25cc)
I recalled how the top status area deals with small screens. This means there is no reason for tabs to get so squished that Downloads hangs off the edge.

GoogleFrog commented:
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/94e091a4-cebf-4d9f-8c20-a73a7c5f9133)
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/c564643c-7405-4847-9b25-05d207e2c61f)
Overall is seems a bit cleaner. The buttons are crisper. The gradients on the main panel and battle rooms seem a bit strong, they get pretty dark towards the bottom right. The number of shades of white used make it feel a bit messy on a few screens.

Personally I don't see what is wrong with decorated borders. Not everything needs to be a flat grey gradient.

There are also a bunch of alignment things to touch up. Eg with exit.
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/cc4993e9-0209-4ba9-a6df-7d9c7550e677)
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/c1631950-2e88-4220-820b-992f7c46caa4)

![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/e39d9f50-ef7f-4379-a0ce-c1dbdd8b6c99)
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/771caaae-22a5-4bfd-b99c-29338203b8a0)

GoogleFrog commented:
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/8b31d018-8c0e-4598-bece-e16cdf5e5f85)
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/b2cc467c-3ab5-45f5-9879-d742b5705200)
This looks better to me. Flat black with a border has better contrast, doesn't overuse grey, and the borders add interest.

GoogleFrog commented:
Minor comments:
- Why are MM battles red? To me that indicates people shouldn't click on them. Surely passworded battles would be red if anything?
- The button behind the minimap has a gradient on it. It probably shouldn't, since it is acting more as a container for non-square maps (like black bars on movies/videos) than a standard button.
- Chat and settings lines don't align when switching between the tabs. They should, I suspect this is preexisting though.

completesomething commented:
New changes, including addressing the majority of the above issues:
* Replaced the bottom link button images in the Welcome/Profile tabs with something less bright and eye-catching.
* Modified the large main windows to have _very_ subtle and smoother gradients. I'd really rather not compromise more than that.
* Reverted the checkboxes to their stable state.
* Normalize padding of main windows.
* Gave tabbar items proper BKs and FGs so that they show better in single pane mode.
* Adjusted battle button colors, swapping red with violet, decreasing opacity of red, and slightly increasing opacity of green/yellow buttons.
* Since the top right buttons are now relationally scaled like in BAR Chobby, lessened the button widths and some `titleWidth`s with `TitleWidthRel` to ensure proper padding.
* Made selected button colors less bright.
* Differentiated selected color of `button` and `highlight_button` to keep the battle room playing/spectating button fairly bright.
* Adjusted loading screen stars to slightly fade out downward.

> Chat text box colour isn't the same as the boxes that hold player names and chat. 
Intentional.

>There are also a bunch of alignment things to touch up.

I've adjusted the main panel padding which should have address the majority of alignment issues. There's still minor mispadding of the Close buttons which is a bit of a different problem, but I don't think it should be blocking.

> Chat and settings lines don't align when switching between the tabs. They should, I suspect this is preexisting though.

This is existing behavior and I don't think it's a problem that should be fixed. The settings tabs are bigger as they're more important to most users.

> Do clickable things always need to be dark like this? 
> The grey in general seems to make things a bit messier/harder to read.
> The button behind the minimap has a gradient on it. 

I feel these particular points are rather subjective and arguable. I don't think they should be blocking issues unless others agree that they are.

completesomething committed:
Add a new Chobby skin + icon refresh

GoogleFrog commented:
Are you force pushing this or something? Removing the commit history makes this unnecessarily hard to pull and keep track of what is going on.

GoogleFrog commented:
> This is existing behavior and I don't think it's a problem that should be fixed. The settings tabs are bigger as they're more important to most users.

The lines, not the tabs at the top.

![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/b1b8ab50-3931-446e-a6ba-6e796bb6734d)

GoogleFrog commented:
Updated preview
![image](https://github.com/ZeroK-RTS/Chobby/assets/1318249/0d533134-bbab-4f79-858f-a782e2c34e38)

GoogleFrog commented:
To revisit this, I don't think the undecorated grey gradient brackground is working. Numerous people think the old border and flat back background looks better so I think we should just stick with it. Besides, the alignment of buttons and panels still needs updating for the new background anyway.

completesomething committed:
Addtional Chobby skin adjustments

completesomething commented:
![skin](https://github.com/user-attachments/assets/002c1df4-c162-4313-b409-8f09b6e78b18)
On request from @Azim-Palmer I've pushed what I've currently have. Still a massive work-in-progress, anything I've not already shown in the Discord likely uses an older skinning and is very likely to be changed.

Known to-dos:
- The battle status fight icon covering the minimap might be a little too much.
- The lobby list battle buttons haven't been updated to the newer style.
- The top bar graphic that shows while in-game hasn't been updated to the newer style.
- The player icon (commander icon) is too complicated and should be simplified.
- The vote buttons haven't been updated to the newer style.
- The scrollbar when horizontally scrolling chat tabs needs to be changed.
- Popup dialogs still use older graphics.
- Play, Load, Start, and Link buttons in lesser used parts of the menu need to updated to the newer style.
- The buttons for the graphics presets need to be changed or padded.
- In some cases the close "X" button doesn't have enough contrast with the background, like in the campaign dialogs.
- The campaign line colors should be adjusted.
- The dropdown button highlights should be adjusted.
- Negative buttons for actually dangerous actions like disabling the lobby should be made red.
